pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -81,6 +83,32 @@ func writeVifFile(buf []byte, pid, name string) error {
 	return nil
 }
 
+func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid string) error {
+	link, err := Handler.LinkByName(podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// get IP address
+	addrList, err := Handler.AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// no ip assigned. ipam disabled
+	if len(addrList) == 0 {
+		return nil
+	}
+
+	err = writeToCachedFile(PodCacheInterface{Iface: iface, PodIP: addrList[0].IP.String()},
+		util.VMIInterfacepath, uid, iface.Name)
```

Comment:
are there any difference between the two locations? or something with ` /var/run/kubevirt-private` why it shouldn't be used? I don't mind changing the location though, just asking for clarity

Reply:
there's a different between the two locations.

```/var/run/kubevirt-private``` is used by virt-handler to store information it needs that it 100% controls.

```/proc/pid/root/var/run/kubevirt-private/``` is used when virt-handler needs to set information in the virt-launcher pod's private directory, so we're reaching into the virt-launcher pod's directory tree, and writing something there.

For this PR, the data we need to cache should be stored in virt-handler's persistent directory. so the directory is correct as it is. 

The reason we can't use the virt-launcher's private directory is because virt-launcher isn't trusted for this information. If virt-launcher's pod was compromised, and a third party could write whatever values they wanted into this cache file, then they could ultimately control what IP people think point to the VMI in question.  This means they could write another IP to the cache file that points to another endpoint which might result in users trying to contact the VMI on an IP that isn't the VMI at all.

hopefully that makes sense. I know the directories here are convoluted.

Reply:
@davidvossel thanks for the detailed explanation!